WordPress/Sniffs/Theme/NoHardCodedUrlsSniff.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * WordPress_Sniffs_Theme_NoHardCodedUrlsSniff.
+ *
+ * WARNING Verify that no hard-coded URLs are found. Manual verification necessary.
+ * An exception is made for the Author URI and (Theme) URI as set in the style.css header
+ * as well as links to wordpress.org. Links in the text of PHP/JS comments should be
+ * excluded from this check.
+ * Ref: https://make.wordpress.org/themes/handbook/review/required/theme-check-plugin/#info
+ * Theme check file covering this rule:
+ * https://github.com/Otto42/theme-check/blob/master/checks/links.php
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @author  Kevin Archibald
+ */
+class WordPress_Sniffs_Theme_NoHardCodedUrlsSniff implements PHP_CodeSniffer_Sniff
+{
+  /**
+   * Returns an array of tokens this test wants to listen for.
+   *
+   * @return array
+   */
+  public function register()
+  {
+      return array(
+        T_STRING,
+        T_CONSTANT_ENCAPSED_STRING,
+        T_INLINE_HTML,
+      );
+  }// end register()
+
+  /**
+   * Processes this test, when one of its tokens is encountered.
+   *
+   * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+   * @param int                  $stackPtr  The position of the current token
+   *                                        in the stack passed in $tokens.
+   *
+   * @return void
+   */
+  public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
+  {
+      $tokens = $phpcsFile->getTokens();
+      $token  = $tokens[$stackPtr];
+
+      /** If the token contains this string, exclude it
+       * ToDo - How can we account for an allowed footer link?
+       * If we require the footer link to be an Author URI or Theme URI it's possible
```

Comment:
We do require the footer link to be either the Author URI or Theme URI. This used to be made clearer, but I don't think it's specified in our guidelines anymore.